backend/endpoints/tests/auth_test.go

```diff
@@ -1,11 +1,93 @@
-package tests
-
-import (
-	"testing"
-)
-
-func TestLogin(t *testing.T) {
-}
-
-func TestLogout(t *testing.T) {
-}
+package tests
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"cms.csesoc.unsw.edu.au/endpoints"
+	"cms.csesoc.unsw.edu.au/endpoints/models"
+
+	"cms.csesoc.unsw.edu.au/database/repositories"
+	. "cms.csesoc.unsw.edu.au/database/repositories/mocks"
+	. "cms.csesoc.unsw.edu.au/endpoints/mocks"
+)
+
+const TEST_EMAIL = "thomas.liang1@student.unsw.edu.au"
+const TEST_PASSWORD = "backendnumberone!"
+
+// Test [endpoints.LoginHandler] succeeds on correct input.
+func TestLogin(t *testing.T) {
+	form := newTestUser()
+	responseRecorder := httptest.NewRecorder()
```

Comment:
nice, TIL this was a thing :P

Reply:
Thanks! Lowkey sick of the signature Go documentation teal already 😩